server/store/setting_store.go

```diff
@@ -84,24 +85,46 @@ func (s *pluginStore) updateDailySummarySettingForUser(userID string, value inte
 		return err
 	}
 
+	stringValue := value.(string)
+	splittedValue := strings.Split(stringValue, " ")
+	timezone := strings.Join(splittedValue[1:], " ")
+
+	found := false
 	for _, dsum := range dsumIndex {
 		if dsum.MattermostUserID == userID {
-			stringValue := value.(string)
-			if stringValue == "true" {
+			found = true
+
+			if splittedValue[0] == "true" {
 				dsum.Enable = true
 				break
 			}
 
-			if stringValue == "false" {
+			if splittedValue[0] == "false" {
 				dsum.Enable = false
 				break
 			}
 
-			dsum.PostTime = value.(string)
+			dsum.PostTime = splittedValue[0]
+			dsum.Timezone = timezone
```

Comment:
This seems pretty hacky to me. What is the reason for needing to accept `interface{}` here?

Maybe make a struct to pass in here instead of doing string parsing. We can still accept `interface{}`, but assert it to be what we expect it to be. This way, the caller can be more specific with what it's passing.

Reply:
There are many considerations here to do it this way.

First of all, this function is called (with a few steps) by the handler, which is agnostic.
The handler passes the context in case of buttons, the option in case of selects. Since this is a select in general, we should receive a string, which is the only value an option can have, and then parse it.

Regarding receiving interface instead of string, is just to make the type conversions where it makes sense, which is here. Otherwise, we would have to make the conversions in the case block on the switch on SetSetting.

We could put all the parsing in one function that returns this struct you are talking about, but I am not sure whether it would improve the readibility.

Reply:
What happens when this value gets more complicated, or we have a more complicated value we have to handle similarly? Is there anyway the origin of the data can tell this function "this is the post time", or "this is the enable flag"? For example, if this store value had more than one boolean, this would not work.

Reply:
We could codify the option content as a JSON string. What do you think?